src/ILLink.RoslynAnalyzer/RequiresAssemblyFilesAnalyzer.cs

```diff
@@ -118,6 +118,12 @@ public override void Initialize (AnalysisContext context)
 					CheckCalledMember (operationContext, eventRef.Member, dangerousPatterns);
 				}, OperationKind.EventReference);
 
+				context.RegisterOperationAction (operationContext => {
+					var delegateCreation = (IDelegateCreationOperation) operationContext.Operation;
+					var target = (IMethodReferenceOperation) delegateCreation.Target;
```

Comment:
This cast doesn't feel safe. the target may be `IAnonymousFunctionOperation` for something like `Action x = () => { };` or can even be a none operation for invalid code.